software/py/vanilla_stats_parser.py

```diff
@@ -0,0 +1,472 @@
+stats_instr_header_format  = ("{:<"+str(name_length)+"}{:>"+str(int_length)+"}{:>"+str(type_length)+"}\n")
+stats_instr_data_format    = ("{:<"+str(name_length)+"}{:>"+str(int_length)+"}{:>"+str(percent_length)+"f}\n")
+stats_stall_header_format  = ("{:<"+str(name_length)+"}{:>"+str(type_length)+"}{:>"+str(type_length)+"}\n")
+stats_stall_data_format    = ("{:<"+str(name_length)+"}{:>"+str(int_length)+"}{:>"+str(percent_length)+"f}\n")
+stats_miss_header_format   = ("{:<"+str(name_length)+"}{:>"+str(type_length)+"}{:>"+str(type_length)+"}{:>"+str(type_length)+"}\n")
+stats_miss_data_format     = ("{:<"+str(name_length)+"}{:>"+str(int_length)+"}{:>"+str(int_length)+"}{:>"+str(float_length)+"f}\n")
+separating_line = ('=' * 90 + '\n')
+
+
+
+# List of instructions, operations and events parsed from vanilla_stats.log
+stats_list  = ["time", "x", "y", "tag", "global_ctr", "cycle"]
+
+instr_list  = ['instr','fadd','fsub','fmul','fsgnj','fsgnjn',
+               'fsgnjx','fmin','fmax','fcvt_s_w','fcvt_s_wu',
+               'fmv_w_x','feq','flt','fle','fcvt_w_s','fcvt_wu_s',
+               'fclass','fmv_x_w','local_ld','local_st',
+               'remote_ld_dram','remote_ld_global','remote_ld_group',
+               'remote_st_dram','remote_st_global','remote_st_group',
+               'local_flw','local_fsw','remote_flw','remote_fsw',
+               'lr','lr_aq','swap_aq','swap_rl','beq','bne',
+               'blt','bge','bltu','bgeu','jalr','jal', 'sll',
+               'slli','srl','srli','sra','srai','add','addi','sub',
+               'lui','auipc','xor','xori','or','ori','and','andi',
+               'slt','slti','sltu','sltiu','mul','mulh','mulhsu',
+               'mulhu','div','divu','rem','remu','fence']
+
+miss_list   = ['icache_miss', 'beq_miss', 'bne_miss', 'blt_miss',
+               'bge_miss', 'bltu_miss', 'bgeu_miss', 'jalr_miss']
+
+stalls_list = ['stall_fp_remote_load','stall_fp_local_load',
+               'stall_depend',
+               'stall_depend_remote_load_dram',
+               'stall_depend_remote_load_global',
+               'stall_depend_remote_load_group',
+               'stall_depend_local_load','stall_force_wb',
+               'stall_ifetch_wait','stall_icache_store',
+               'stall_lr_aq','stall_md','stall_remote_req','stall_local_flw']
+
+
+
+class VanillaStatsParser:
+
+  # default constructor
+  def __init__(self, manycore_dim_y, manycore_dim_x, per_tile_stat):
+
+    self.manycore_dim_y = manycore_dim_y
+    self.manycore_dim_x = manycore_dim_x
+    self.manycore_dim = manycore_dim_y * manycore_dim_x
+    self.per_tile_stat = per_tile_stat
+
+
+    self.max_tile_groups = 1024
+    self.num_tile_groups = 0
+
+    self.timing_start_list = [0] * self.max_tile_groups 
+    self.timing_end_list =   [0] * self.max_tile_groups
+#    self.timing_stat = []
+
+    self.total_execution_time = 0
+    self.total_instr_cnt = 0
```

Comment:
if you want counters, there's a cool dict subclass called Counter (`from collections import Counter`)